src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -6973,12 +7028,11 @@ extends System.Object
             var source = @"
 public record B : A {
 }";
-            var comp = CreateCompilationWithIL(new[] { source, IsExternalInitTypeDefinition }, ilSource: ilSource, parseOptions: TestOptions.Regular9);
-            comp.VerifyEmitDiagnostics(
-                // (2,15): error CS0239: 'B.ToString()': cannot override inherited member 'A.ToString()' because it is sealed
-                // public record B : A {
-                Diagnostic(ErrorCode.ERR_CantOverrideSealed, "B").WithArguments("B.ToString()", "A.ToString()").WithLocation(2, 15)
-                );
+            var comp = CreateCompilationWithIL(
+                new[] { source, IsExternalInitTypeDefinition },
+                ilSource: ilSource,
+                parseOptions: usePreview ? TestOptions.RegularPreview : TestOptions.Regular9);
```

Comment:
>TestOptions.Regular9 [](start = 72, length = 20)

I think we should report a language version error for `Regular9` because previous compiler wouldn't be able to compile this code and targeting that version should prevent a build break like that when compiler is switched back.

Reply:
Sorry, maybe I'm a bit slow (it's 3 a.m. here), but I don't get it... Could you elaborate?

Reply:
>> I think we should report a language version error for Regular9 because previous compiler wouldn't be able to compile this code and targeting that version should prevent a build break like that when compiler is switched back.

> Sorry, maybe I'm a bit slow (it's 3 a.m. here), but I don't get it... Could you elaborate?

There is a requirement - If compiling some code with a particular language version succeeds, then compiling the same code by using an old version of the compiler that added support for that language version should also succeed and result should be equivalent. This is an error scenario in the previous version of the compiler. Therefore, when we compile this code with language version C# 9, we should report an error, saying that the language version should be bumped up. 

---
In reply to: [602653525](https://github.com/dotnet/roslyn/pull/52029#discussion_r602653525) [](ancestors = 602653525)

Reply:
So, just to make sure I understand it right... If language version is 9, it should:
- not emit a `ToString()` override (since the method is sealed in the base type)
- emit a `ERR_FeatureInPreview` diagnostic

Is that correct?

I just worry that the error will be a bit confusing. The user would just be trying to inherit a record with a sealed ToString (e.g. from another assembly compiled with C# 10), but the error will say that "sealed ToString is records" is a preview feature.

Reply:
>So, just to make sure I understand it right... If language version is 9, it should:
>- not emit a ToString() override (since the method is sealed in the base type)
>- emit a ERR_FeatureInPreview diagnostic
>
>Is that correct?
>
>I just worry that the error will be a bit confusing. The user would just be trying to inherit a record with a sealed ToString (e.g. from another assembly compiled with C# 10), but the error will say that "sealed ToString is records" is a preview feature.

If the simple error will be confusing, we should consider adding a dedicated more descriptive error, which will request to bump the language version. For example: "Inheriting from a class with a sealed 'Object.ToString' is not supported in C# 9. Please use language version 'preview' or greater."

---
In reply to: [602872811](https://github.com/dotnet/roslyn/pull/52029#discussion_r602872811) [](ancestors = 602872811)

Reply:
OK, I addressed that in commit 7dcf8150e63. (also added the new error message)